Suggestions

SiverDX opened the pull request:
Refactoring / logic adjustment suggestions

SiverDX committed:
Adjust some mixins
- replace some redirects
- simplify things

JohnSmith474Auth commented on common/src/main/java/johnsmith/enchantingoverhauled/mixin/block/PowderSnowBlockMixin.java:
Status: Accepted

JohnSmith474Auth commented on common/src/main/java/johnsmith/enchantingoverhauled/mixin/component/type/ItemEnchantmentsMixin.java:
Status: Accepted

JohnSmith474Auth commented on common/src/main/java/johnsmith/enchantingoverhauled/mixin/enchantment/EnchantmentMixin.java:
Status: Rejected
Reasons:
1. Unnecessary bounds check. Config value is bound between 1 and 254 inclusive.
2. File is pending removal.

SiverDX replied:
I added it so you can remove the max level change

Since I can't just set to to 10 (it will cause all enchantments to go up to that level)

JohnSmith474Auth commented on common/src/main/java/johnsmith/enchantingoverhauled/mixin/menu/AnvilMenuMixin.java:
Status: Needs Changes.
Reason: Ambiguous file name, see [fabric/src/main/java/johnsmith/enchantingoverhauled/mixin/menu/AnvilMenuMixin.java](https://github.com/JohnSmith474Mods/Enchanting-Overhauled/pull/7/files#diff-b2b7567606e03564bfdfc18c76b4a60fdaefa75c7cd7740f3e420394b6d47c66).

JohnSmith474Auth commented on fabric/src/main/java/johnsmith/enchantingoverhauled/mixin/menu/FabricAnvilMenuMixin.java:
Status: Needs Changes
Reason: Ambiguous file name

JohnSmith474Auth commented on fabric/src/main/resources/enchanting_overhauled.fabric.mixins.json:
Status: Accepted, under the condition that [fabric/src/main/java/johnsmith/enchantingoverhauled/mixin/menu/AnvilMenuMixin.java](https://github.com/JohnSmith474Mods/Enchanting-Overhauled/pull/7/files#diff-b2b7567606e03564bfdfc18c76b4a60fdaefa75c7cd7740f3e420394b6d47c66) is disambiguated.

JohnSmith474Auth commented on neoforge/src/main/java/johnsmith/enchantingoverhauled/events/EventHandler.java:
Status: Accepted, under the condition that [fabric/src/main/java/johnsmith/enchantingoverhauled/mixin/menu/AnvilMenuMixin.java](https://github.com/JohnSmith474Mods/Enchanting-Overhauled/pull/7/files#diff-b2b7567606e03564bfdfc18c76b4a60fdaefa75c7cd7740f3e420394b6d47c66) is disambiguated.

JohnSmith474Auth reviewed (changes requested):
Requested changes:
1. Remove changes to `common/src/main/java/johnsmith/enchantingoverhauled/mixin/enchantment/EnchantmentMixin.java` since it is pending removal.
2. Change `**fabric**/src/main/java/johnsmith/enchantingoverhauled/mixin/menu/AnvilMenuMixin.java` to a name which will not collide with `**common**/src/main/java/johnsmith/enchantingoverhauled/mixin/menu/AnvilMenuMixin.java`

SiverDX commented:
(Technically also WIP, therefor still a draft)

SiverDX committed:
changes
- adjust fabric mixin name
- allow value of 0 for max enchantment level bound (to disable override)
- (changed some other usages to 'getMaxLevel' since the mixin to that method will refer to the config if relevant)

CreativelyJake commented:
Hi! new to using github, but if this is for suggestions for the mod: I'd like to be able to turn off the EXP changes, so that I can use the EXP scaling of a mod like "Linear Levels" alongside EO. Currently, this mod overrides Linear Levels no matter what

JohnSmith474Auth commented:
> Hi! new to using github, but if this is for suggestions for the mod: I'd like to be able to turn off the EXP changes, so that I can use the EXP scaling of a mod like "Linear Levels" alongside EO. Currently, this mod overrides Linear Levels no matter what

Please open a new ticket for issues and suggestions.

I have already implemented the changes you requested, but I am currently short on time and I ran into issues with my config management solution which I am currently trying to export into its own library. So it will be some time until this feature releases.
